packages/lesswrong/components/users/CrosspostLoginPage.tsx

```diff
@@ -0,0 +1,110 @@
+import React, { useState } from "react";
+import { Components, registerComponent } from "../../lib/vulcan-lib";
+import Button from "@material-ui/core/Button";
+import { useCurrentUser } from "../common/withUser";
+import { useGetParameter } from "../hooks/useGetParameter";
+import { forumHeaderTitleSetting } from "../common/Header";
+import { forumTypeSetting } from "../../lib/instanceSettings";
+import { gql, useMutation } from "@apollo/client";
+
+const styles = (theme: ThemeType): JssStyles => ({
+  root: {
+    display: "flex",
+    flexDirection: "column",
+    alignItems: "center",
+  },
+  heading: {
+    display: "flex",
+    flexDirection: "row",
+    alignItems: "center",
+    marginTop: "1.5em",
+  },
+  headingText: {
+    fontSize: 20,
+    margin: 0,
+  },
+  buttonContainer: {
+    marginTop: "1.5em",
+  },
+  error: {
+    color: theme.palette.error.main,
+    marginTop: 16,
+  },
+});
+
+const connectCrossposterMutation = gql`
+  mutation connectCrossposter($token: String) {
+    connectCrossposter(token: $token)
+  }
+`;
+
+const CrosspostLoginPage = ({classes}: {
+  classes: ClassesType,
+}) => {
+  const [mutate, loading] = useMutation(connectCrossposterMutation, {errorPolicy: "all"});
+  const [error, setError] = useState<string | null>(null);
+  const currentUser = useCurrentUser();
+  const token = useGetParameter("token");
+  const hasLogo = forumTypeSetting.get() === "EAForum";
+
+  const onConfirm = async () => {
+    if (!currentUser) {
+      throw new Error("Can't connect crosspost account whilst logged out");
+    }
+    const result = await mutate({
+      variables: {token},
+    });
+    if (result?.data?.connectCrossposter === "success") {
+      setError(null);
+      window.close();
+    } else {
+      setError("Failed to connect accounts");
+    }
+  }
+
+  const {WrappedLoginForm, SiteLogo, Loading, Typography} = Components;
+
+  return (
+    <div className={classes.root}>
+      <div className={classes.heading}>
+        {hasLogo && <SiteLogo />}
+        <Typography variant="title" className={classes.headingText}>
+          {forumHeaderTitleSetting.get()}
+        </Typography>
+      </div>
+      {error && <div className={classes.error}>Error: {error}</div>}
+      {currentUser
+        ? (
+          <>
+            <Typography variant="body2">
+              Logged in as {currentUser.displayName}
+            </Typography>
+            <div className={classes.buttonContainer}>
+              {loading?.loading || (loading?.called && !loading?.error)
```

Comment:
is the `loading` field in `loading` (speaking of which: 😬) a boolean?  If not, can use `??` instead of `||`, which is narrower.  (If it's a boolean, I assume the `||` is deliberate for coalescing on `false` values, not just `null | undefined`.)

Reply:
I second your 😬. `loading.loading` is indeed as boolean, so I think the `||` is correct.

Reply:
Ok, sounds good.